File: _includes/code/llms-txt/typescript/aggregations.ts
// llms.txt snippet: aggregations. Section "Python / TypeScript > Aggregations".
import weaviate, { vectors, dataType } from 'weaviate-client';

const client = await weaviate.connectToLocal();
await client.collections.delete('Movie');
await client.collections.create({
  name: 'Movie',
  vectorizers: vectors.text2VecOllama({ apiEndpoint: 'http://ollama:11434', model: 'nomic-embed-text' }),
  properties: [
    { name: 'title', dataType: dataType.TEXT },
    { name: 'genre', dataType: dataType.TEXT },
    { name: 'rating', dataType: dataType.NUMBER },
  ],
});
const movies = client.collections.use('Movie');
await movies.data.insertMany([
  { title: 'The Matrix', genre: 'Science Fiction', rating: 8.7 },
  { title: 'Spirited Away', genre: 'Animation', rating: 8.6 },
  { title: 'Blade Runner', genre: 'Science Fiction', rating: 8.1 },
]);

// START llms_aggregations
// Total object count
const total = (await movies.aggregate.overAll()).totalCount;

// Numeric metric over a property (mean rating)
const ratingAgg = await movies.aggregate.overAll({
  returnMetrics: movies.metrics.aggregate('rating').number(['mean']),
});

// Group object counts by a property
const byGenre = await movies.aggregate.groupBy.overAll({ groupBy: { property: 'genre' } });
// END llms_aggregations

if (total !== 3) throw new Error(`expected 3 objects, got ${total}`);
if (byGenre.length !== 2) throw new Error('expected 2 genre groups');
await client.collections.delete('Movie');
await client.close();

File: _includes/code/llms-txt/python/aggregations.py
"""llms.txt snippet: aggregations. Section "Python / TypeScript > Aggregations"."""
import os
import weaviate
from weaviate.classes.init import Auth
from weaviate.classes.config import Configure, Property, DataType

client = weaviate.connect_to_weaviate_cloud(
    cluster_url=os.environ["WEAVIATE_URL"],
    auth_credentials=Auth.api_key(os.environ["WEAVIATE_API_KEY"]),
)
client.collections.delete("Movie__AggsPy")
client.collections.create(
    "Movie__AggsPy",
    vector_config=Configure.Vectors.text2vec_weaviate(),
    properties=[
        Property(name="title", data_type=DataType.TEXT),
        Property(name="genre", data_type=DataType.TEXT),
        Property(name="rating", data_type=DataType.NUMBER),
    ],
)
movies = client.collections.use("Movie__AggsPy")
movies.data.insert_many([
    {"title": "The Matrix", "genre": "Science Fiction", "rating": 8.7},
    {"title": "Spirited Away", "genre": "Animation", "rating": 8.6},
    {"title": "Blade Runner", "genre": "Science Fiction", "rating": 8.1},
])

# START llms_aggregations
from weaviate.classes.aggregate import GroupByAggregate, Metrics

# Total object count
total = movies.aggregate.over_all(total_count=True).total_count

# Numeric metric over a property (mean rating)
res = movies.aggregate.over_all(
    return_metrics=Metrics("rating").number(mean=True),
)

# Group object counts by a property
groups = movies.aggregate.over_all(group_by=GroupByAggregate(prop="genre")).groups
# END llms_aggregations

assert total == 3
assert res.properties["rating"].mean is not None
assert len(groups) == 2
client.collections.delete("Movie__AggsPy")
client.close()
